src/sentry/static/sentry/app/utils/errorHandler.jsx

```diff
@@ -2,17 +2,32 @@ import React from 'react';
 import RouteError from 'app/views/routeError';
 
 export default function errorHandler(Component) {
-  const originalRender = Component.prototype.render;
-  Component.prototype.render = function() {
-    try {
-      return originalRender.apply(this, arguments);
-    } catch (err) {
-      /*eslint no-console:0*/
+  class ErrorHandler extends React.Component {
+    static getDerivedStateFromError(error) {
       setTimeout(() => {
-        throw err;
+        throw error;
       });
-      return <RouteError error={err} component={this} />;
+
+      // Update state so the next render will show the fallback UI.
+      return {
+        hasError: true,
+        error,
+      };
+    }
+
+    state = {
+      hasError: false,
```

Comment:
Is this a bit redundant? We know error state with `state.error`

Reply:
I initially thought that the `error` thrown would always be truthy; but interestingly, that isn't the case since JS lets you throw anything e.g. `throw null`. 

Here's a sample react app showcasing that `error` can be a falsy value received by `static getDerivedStateFromError(error)`: https://codesandbox.io/s/mdq0z?fontsize=14

Reply:
Hah, that's an additional issue, we should never be throwing `null`, but this is 👍